File: src/main/scala/com/strings/leetcode/string/Problem_657_judgeCircle.scala
package com.strings.leetcode.string

object Problem_657_judgeCircle {

  def main(args: Array[String]): Unit = {
    val moves = "LL"
    println(judgeCircle(moves))
  }


  def judgeCircle(moves: String): Boolean = {
    var num_vertical = 0
    var num_horizontal = 0
    for(c <- moves.toCharArray){
      if(c == 'R') num_horizontal = num_horizontal + 1
      if(c == 'L') num_horizontal = num_horizontal - 1
      if(c == 'U') num_vertical = num_vertical + 1
      if(c == 'D') num_vertical = num_vertical - 1
    }
    if(scala.math.abs(num_horizontal) + scala.math.abs(num_vertical)== 0) true
    else false
  }

}

File: src/main/java/string/Problem_657_judgeCircle.java
package string;

public class Problem_657_judgeCircle {

    public boolean judgeCircle(String moves){
        int num_vertical = 0;
        int num_horizontal = 0;
        for(char c: moves.toCharArray()){
            if(c == 'R') num_horizontal ++;
            if(c == 'L') num_horizontal --;
            if(c == 'U') num_vertical ++;
            if(c == 'D') num_vertical --;
        }
        if(Math.abs(num_horizontal) + Math.abs(num_vertical)== 0)
            return true;
        else
            return false;
    }
}
